### PR TITLE
DROOLS-4973: Adding additional test cases for FEEL expressions

### PR DESCRIPTION
**Pull request by yesamer**
@jomarko @danielezonca 

Added test cases for case reported in https://issues.redhat.com/browse/DROOLS-4973

### CODE REVIEW

**Review by danielezonca — APPROVED** [review]

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java [hunk 1]
```diff
@@ -78,6 +78,27 @@ public void evaluateUnaryExpression() {
...
+        Map<String, Object> secondMap = new HashMap<>();
+        secondMap.put("Price", new BigDecimal(3300));
+        secondMap.put("Name", "CAR");
+        List<Map<String, Object>> context = Arrays.asList(firstMap, secondMap);
+        assertTrue(expressionEvaluator.evaluateUnaryExpression(new TextNode("?=[{Price: 2000,Name:\"PC\"}, {Price:3300,Name:\"CAR\"}]").toString(), context, List.class));
+        assertFalse(expressionEvaluator.evaluateUnaryExpression(new TextNode("?=[{Price: 2001,Name:\"PC\"}, {Price:3301,Name:\"CAR\"}]").toString(), context, List.class));
+        assertFalse(expressionEvaluator.evaluateUnaryExpression(new TextNode("?=[{Price: 2000, Name:\"PCA\"}, {Price:3300,Name:\"CARE\"}]").toString(), context, List.class));
+        assertFalse(expressionEvaluator.evaluateUnaryExpression(new TextNode("?=[{Pric: 2000, Name:\"PC\"}, {Price:3300,Names:\"CARE\"}]").toString(), context, List.class));
+        /* Different order: Failure */
+        assertFalse(expressionEvaluator.evaluateUnaryExpression(new TextNode("?=[{Price:3300,Name:\"CARE\"}, {Price: 2000,Name:\"PC\"}]").toString(), context, List.class));
```
Comment (jomarko) [inline drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java]: Also different value, not just order, wee need to fix

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java [hunk 2]
```diff
@@ -78,6 +78,27 @@ public void evaluateUnaryExpression() {
...
+        /* IN operator */
+        assertTrue(expressionEvaluator.evaluateUnaryExpression(new TextNode("{Price: 2000,Name:\"PC\"} in ?").toString(), context, List.class));
+        assertTrue(expressionEvaluator.evaluateUnaryExpression(new TextNode("{Price: 2000,Name:\"PC\"} in ?").toString(), context, List.class));
```
Comment (jomarko) [inline drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java]: same lines

**Review by jomarko — COMMENTED** [review]
Maybe we cound introduce constant for each substring "{Price: 2000,Name:\"PC\"}" and then just conjuct them in constructor of TextNode.

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java (hunk 1 above)
Comment (yesamer) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java]: @jomarko Done, thank you!

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java (hunk 2 above)
Comment (yesamer) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java]: @jomarko Done, thank you!

**Review by jomarko — APPROVED** [review]
Thank you

**Pull request merged**